userguide/tutorials/usage_plugin.adoc

```diff
@@ -0,0 +1,31 @@
+= Developing a Usage Plugin
+
+[[overview]]
+== Overview
+
+A usage plugin implements the https://github.com/killbill/killbill-plugin-api/blob/master/usage/src/main/java/org/killbill/billing/usage/plugin/api/UsagePluginApi.java[UsagePluginApi]. This plugin can be used to collect usage statistics that are collected by a separate application. This application may have specialized knowledge of how to measure specific types of resource units, such as cellphone minutes.
+
+
+=== UsagePluginApi
+
+Developers of usage plugins need to implement the following APIs:
+
+* `getUsageForAccount`: This method gets usage data for a specified account. 
+* `getUsageForSubscription`: This method gets usage data for a particular subscription.
+
+Each of the APIs take the following parameters:
+
+* `startDate`: The beginning of the date range to be queried
+* `endDate`: The end of the date range to be queried
+* `context`: Provides the full `UsageContext` associated with this query, including the account ID.
+* `properties`: A list of `PluginProperty` items that are passed from the client making the API call to the plugin.
+
+In addition,  the `getUsageForAccount` API includes a `subscriptionId` parameter.
```

Comment:
Shouldn't this be `getUsageForSubscription`?

Reply:
Yes, this needs to be fixed.

Reply:
typo corrected.